lib/nmatrix/nmatrix.rb

```diff
@@ -90,6 +90,62 @@ def size(shape)
       shape = [shape,shape] unless shape.is_a?(Array)
       (0...shape.size).inject(1) { |x,i| x * shape[i] }
     end
+
+    # Make N-D coordinate arrays for vectorized evaluations of
+    # N-D scalar/vector fields over N-D grids, given N one-dimensional
+    # coordinate arrays arrs. N > 1.
+    #
+    # call-seq:
+    #     meshgrid(arrs) -> Array of NMatrix
+    #     meshgrid(arrs, options) -> Array of NMatrix
+    #
+    # * *Arguments* :
+    #   - +arrs+ -> Array of N one-dimensional coordinate arrays (Array or NMatrix)
+    #   - +options+ -> Hash with options (:sparse and :ij both Boolean, false by default)
+    # * *Returns* :
+    #   - Array of N N-D NMatrixes
+    # * *Examples* :
+    #     x, y = NMatrix::meshgrid([[1, 2, 3], [4, 5]])
+    #     x.to_a #<= [[1, 2, 3], [1, 2, 3]]
+    #     y.to_a #<= [[4, 4, 4], [5, 5, 5]]
+    #
+    # * *Using* *options* :
+    #
+    #     x, y = NMatrix::meshgrid([[1, 2, 3], [4, 5]], sparse: true)
+    #     x.to_a #<= [[1, 2, 3]]
+    #     y.to_a #<= [[4], [5]]
+    #
+    #     x, y = NMatrix::meshgrid([[1, 2, 3], [4, 5]], ij: true)
+    #     x.to_a #<= [[1, 1], [2, 2], [3, 3]]
+    #     y.to_a #<= [[4, 5], [4, 5], [4, 5]]
+    def meshgrid(arrs, options = {})
+      raise(ArgumentError, 'Expected at least 2 arrays.') if arrs.size < 2
+      mats = arrs.map do |arr|
+        case arr
+        when NMatrix
+          arr
+          raise(ArgumentError, 'Expected one-dimensional arrays.') if arr.effective_dim != 1
```

Comment:
IMO you should just flatten the matrix and use each element instead of enforcing single dimensional matrices.

Octave and numpy both do that:
``` octave
[xa,xb]=meshgrid([[1,2,3],[1,2,3]],[4,5])
xa =

   1   2   3   1   2   3
   1   2   3   1   2   3

xb =

   4   4   4   4   4   4
   5   5   5   5   5   5
```

numpy:
``` python
In[63]: y = np.array([ 0.,  1.])
In [64]: x = np.array([[0.0, 0.5, 1.0],[0.0, 0.5, 1.0]])
In [65]: xv, yv = meshgrid(x, y)
In [66]: xv
Out[66]: 
array([[ 0. ,  0.5,  1. ,  0. ,  0.5,  1. ],
       [ 0. ,  0.5,  1. ,  0. ,  0.5,  1. ]])
In [67]: yv
Out[67]: 
array([[ 0.,  0.,  0.,  0.,  0.,  0.],
       [ 1.,  1.,  1.,  1.,  1.,  1.]])
```

Seems legit to implement it here as well. @mohawkjohn thoughts?